src/app/Archives/ArchiveUploadModal.tsx

```diff
@@ -89,22 +141,25 @@ export const ArchiveUploadModal: React.FunctionComponent<ArchiveUploadModalProps
       reset();
       props.onClose();
     }
-  }, [uploading, setShowCancelPrompt, reset]);
+  }, [uploading, setShowCancelPrompt, reset, props.onClose]);
 
   const handleSubmit = React.useCallback(() => {
     if (!uploadFile) {
       notifications.warning('Attempted to submit JFR upload without a file selected');
       return;
     }
     setUploading(true);
-    context.api.uploadRecording(uploadFile, abort.signal).pipe(first()).subscribe(handleClose, reset);
-  }, [context.api, notifications, setUploading, uploadFile, abort, handleClose, reset]);
+    context.api
+      .uploadRecording(uploadFile, getFormattedLabels(), abort.signal)
+      .pipe(first())
+      .subscribe(handleClose, reset);
+  }, [context.api, notifications, setUploading, uploadFile, abort.signal, handleClose, reset, getFormattedLabels]);
```

Comment:
Should `context` be a dep?

Reply:
I think we just use `context.api` so not sure if it is necessary. But I will run with eslint to check.

Reply:
Looked like the good way is to have specific ref to those deps (i.e. context.api)  instead as a whole (i.e. context) following eslint checks.

Reply:
Sounds good, then in the future those should be removed in many React.useCallback and React.useEffect deps, haha